src/cgame/cg_rocket_draw.cpp

```diff
@@ -1029,6 +1029,34 @@ class WallwalkElement : public HudElement
 	bool isActive;
 };
 
+class SprintElement : public HudElement
+{
+public:
+	SprintElement( const Rocket::Core::String& tag ) :
+			HudElement( tag, ELEMENT_HUMANS ),
+			isActive( false ) {}
+
+	void DoOnUpdate()
+	{
+		bool sprinting = cg.snap->ps.stats[ STAT_STATE ] & SS_SPEEDBOOST;
+		if ( ( sprinting && !isActive ) || ( !sprinting && isActive ) )
```

Comment:
`sprinting != isActive`

Reply:
I admit I shamelessly copy-pasted code from wallwalking. Thus I applied your remarks to both (there's probably a better way to implement that, though).